src/rgw/rgw_rest_swift.cc

```diff
@@ -1660,7 +1714,21 @@ void RGWInfo_ObjStore_SWIFT::list_swift_data(Formatter& formatter,
 
   string ceph_version(CEPH_GIT_NICE_VER);
   formatter.dump_string("version", ceph_version);
-  formatter.dump_int("max_meta_name_length", 81);
+
+  const size_t meta_name_limit = g_conf->osd_max_attr_name_len
+    - strlen(RGW_ATTR_PREFIX RGW_AMZ_META_PREFIX);
+  formatter.dump_int("max_meta_name_length", meta_name_limit);
+
+  const size_t meta_value_limit = g_conf->osd_max_attr_size;
```

Comment:
Again not sure where to put this - but this really should have a default value (ocata swift advertises 256).  Since this is an "osd" value and applies to all rados attributes, setting this to small values causes other interesting bits of rados to start breaking (and fixing that means restarting all osds). From an operational standpoint for rgw, I think the only sane way to set this is as an rgw specific value and have it set to a separate possibly lower value than the osd.  So perhaps this should just be its own separate configuration item.

Reply:
didn't get 

> setting this to small values causes other interesting bits of rados to start breaking (and fixing that means restarting all osds).

The usage pattern of `osd_max_attr_size` in RadosGW assumed setting the option only in the RGW-dedicated section of `ceph.conf`. Does the mentioned problem occur then? If so, we might have an  issue in OSDs as a misbehaving RADOS client shouldn't cause the need to restart it, I believe.

Reply:
If I move this to the rgw only section I get more desirable behavior yes.

Note: naive people looking at an option that starts with "osd_" are likely to think it matters to the osd.  And, oh it does, but not in a way that's useful to swift.

I'd prefer to see this be its own independent configuration name starting with "rgw_".

According to the s3 documentation, http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-metadata , there is also a s3 metadata size limit.  I don't think we want to enforce that limit, but it appears to me if we advertise a default 2K per-meta value limit, we can please the tempest tests, provide a perfectly defensible swift size limit, and an equally defensible s3 experience.

Reply:
Got it. Renaming the configurables to `rgw_`.